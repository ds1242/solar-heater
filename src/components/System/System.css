img {
    width: 60vw;
    height: 65vh;
    margin-left: auto;
    margin-right: auto;
}

#startButton {
    width: 25%;
    padding: 10px;
    margin: 10px;
}

#imageRow {
    padding: 10px
}

#infoSection {
    padding: 30px;
    text-align: center;
}