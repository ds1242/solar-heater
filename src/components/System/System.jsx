import React, { useState,  } from 'react';
import { Container, Row, Col, Button } from 'react-bootstrap';
import pump from '../../assets/pump.png';
import sun from '../../assets/sun.png';
import solarPanel from '../../assets/solar_panel.png';
import storageTank from '../../assets/storage_tank.png';

import systemInitial from '../../assets/initialSystemState.png';
import systemHeat1 from '../../assets/systemHeatingBegin.png';
import systemHeat2 from '../../assets/systemHeating2.png';
import tankHeat1 from '../../assets/tankHeating1.png';
import tankHeat2 from '../../assets/tankHeating2.png';
import tankHeat3 from '../../assets/tankHeating3.png';
import systemFinalState from '../../assets/systemFinalState.png';

function System() {
    let systemHeat = [systemInitial, systemHeat1, systemHeat2, tankHeat1, tankHeat2, tankHeat3, systemFinalState]
    
    let interval = null;
    
    const [imageSource, setImageSource] = useState(systemHeat[0])
    // console.log(imageState.activeIndex)
    const changePicture = (event) => {
        event.preventDefault();
        let newIndex = 0

        interval = setInterval(() => {
            setImageSource(systemHeat[newIndex])
            console.log(newIndex)

            newIndex = newIndex  + 1
            if(newIndex === 7) {
                clearInterval(interval);
            }
        }, 1000)
        // setImageSource(sun);
    }
    
    return (
        <Container>
            <Row>
                
                <Col>
                    <img src={ imageSource } />
                </Col>
                
            </Row>
            <Row>
                <Button onClick={changePicture}>Click</Button>
                
            </Row>
        </Container>
    )
};

export default System;