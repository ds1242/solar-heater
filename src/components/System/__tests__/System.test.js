import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import System from '../System';

afterAll(cleanup);

describe('System component', () => {
    it('renders', () => {
        render(<System />);
    });

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<System />);

        expect(asFragment()).toMatchSnapshot();
    });
});