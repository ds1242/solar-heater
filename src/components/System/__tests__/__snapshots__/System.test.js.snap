// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`System component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="justify-content-center row"
    >
      <button
        class="btn btn-primary"
        id="startButton"
        type="button"
      >
        Click Here For A High Level Explanation
      </button>
    </div>
    <div
      class="justify-content-center align-items-center row"
    >
      <div
        class="col"
        id="infoSection"
      >
        <h3
          class="justify-content-center"
        >
          Sensor in the tank detects the temperature of the water has dropped
        </h3>
      </div>
    </div>
    <div
      class="justify-content-center row"
      id="imageRow"
    >
      <div
        class="col"
      >
        <img
          alt="images of system heating water and adding it to the tank"
          src="initialSystemState.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
